lib/global-styles.php

```diff
@@ -171,22 +187,50 @@ function gutenberg_experimental_global_styles_get_theme() {
  * @return string CSS rule.
  */
 function gutenberg_experimental_global_styles_resolver( $global_styles ) {
```

Comment:
I'm thinking instead of the automatic CSS variables generation we should rely on something very similar to this https://github.com/WordPress/gutenberg/pull/21428/files#diff-d30688729dfd9f0bba44ec6f9bfe6c57R43

Reply:
I'm eager to experiment with that approach! I'm thinking this PR addresses a different problem (theme.json shape, theme.json to CSS conversion) and the exploration about CSS variables or not merits its own PR.

Reply:
I've come round this and now this PR doesn't use CSS Custom Properties for styling blocks, but just normal properties. Note that the example still uses CSS vars for pulling the values from presets, I think that's worthwhile.